src/components/modal/Modal.jsx

```diff
@@ -0,0 +1,71 @@
+// @flow strict
+
+import * as React from 'react';
+import TopLayer from '../toplayer/TopLayer';
+import Overlay from '../overlay/Overlay';
+
+const OVERLAY_CLASS_NAME = 'sg-overlay';
+
+type PropsType = {
```

Comment:
we should think about these props, I believe most of them aren't needed

Reply:
I guess we should think if our goal here is to be backward-compatible or to provide the best API

Reply:
exactly, I think modal is not widely used because of this complex API

Reply:
I agree with you @bartoszlorek  @maciekbb . Moving this component to the SG made me think about how many props are just passed down to the `TopLayer` and many of them are not working in modal in expected way (speaking gently). My initial idea was to keep the same interface for compatibility but if you're not happy from current implementation of Modal this is great case to improve that.